### PR TITLE
Add MSBuild support for AnalyzerDependency items

### PR DESCRIPTION
**Pull request by tmeschter**
This commit updates the Vbc and Csc build tasks (and related targets
files) to consume AnalyzerDependency items. They are passed to the
command-line compilers using the "/analyzerdependency" switch, or on to
a build host through a new IAnalyzerDependencyHostObject COM interface.

### CODE REVIEW

**Comment by tmeschter** [comment]
@srivatsn @mavasani @JohnHamby @jmarolf @shyamnamboodiripad @heejaechang @agocke Could you take a look at this, please?

File: src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets [hunk 1]
```diff
@@ -70,6 +70,7 @@
           AdditionalFiles="@(AdditionalFiles)"
           AllowUnsafeBlocks="$(AllowUnsafeBlocks)"
           Analyzers="@(Analyzer)"
+          AnalyzerDependencies="@(AnalyzerDependency)"
```
Comment (tmeschter) [inline src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets]: @agocke Should we be capturing Analyzers and AnalyzerDependencies as inputs (that is, in the Inputs attribute of the Target element, at the top of the file)? It really seems like we should...

Comment (tmeschter) [inline src/Compilers/Core/MSBuildTask/Microsoft.CSharp.Core.targets]: I've filed #1753 to track this issue.

**Comment by tmeschter** [comment]
@srivatsn @mavasani @JohnHamby @jmarolf @shyamnamboodiripad @heejaechang @agocke Ping?

**Comment by heejaechang** [comment]
:+1:

**Pull request merged**